reframe/core/schedulers/slurm.py

```diff
@@ -591,6 +615,10 @@ def is_available(self):
         return all([self._states == {'IDLE'}, self._partitions,
                     self._active_features, self._states])
 
+    def under_maintenance(self):
+        return all([self._states == {'MAINT'}, self._partitions,
+                    self._active_features, self._states])
```

Comment:
Couldn't we have a more generic method that would pick nodes that are in a specific state? I don't like this very much. It's too verbose and very specific.

Reply:
I wrote it like that so that it's more generic if we need to add different nodes that might use a different convention for maintenance. I forgot to add it to the abc `Node` class. Should I keep it like that?

Reply:
I think the most generic would be to use the argument passed `--flex-alloc-nodes` to denote the state. You just convert that to uppercase. So for example, one could also write `--flex-alloc-nodes=maint+drain`

Reply:
Yes it can be like that. I will have to change the logic which currently relies on sets and uses `+` as a separator for the list of states os a node.

Reply:
It can still rely on those I guess, we can discuss this offline.